Sources/PerfectLib/Dir.swift

```diff
@@ -56,19 +56,17 @@ public struct Dir {
 	public func create(perms: Int = Int(S_IRWXG|S_IRWXU|S_IRWXO)) throws {
 		let pth = realPath()
 		var currPath = pth.begins(with: "/") ? "/" : ""
-
-		for component in pth.pathComponents {
-			if component != "/" {
-				currPath += component
-				if !exists(currPath) {
-					let res = mkdir(currPath, mode_t(perms))
-					guard res != -1 else {
-						try ThrowFileError()
-					}
-				}
-				currPath += "/"
-			}
-		}
+        for component in pth.pathComponents where component != "/" {
+            currPath += component
+            guard !exists(currPath) else {
+                currPath += "/"
```

Comment:
`currPath += "/"` should be in a `defer` since it gets executed regardless if the current path exists (at least, that's what the semantics of the code were pre-pr).